pkg/platform/model/buildplanner.go

```diff
@@ -345,3 +349,17 @@ func (bp *BuildPlanner) GetBuildExpression(owner, project, commitID string) (*bu
 
 	return expression, nil
 }
+
+// processBuildPlannerError will check for special error types that should be
+// handled differently. If no special error type is found, the fallback message
+// will be used.
+func processBuildPlannerError(bpErr error, fallbackMessage string) error {
+	graphqlErr := &graphql.GraphErr{}
+	if errors.As(bpErr, graphqlErr) {
+		code, ok := graphqlErr.Extensions[codeExtensionKey].(string)
+		if ok && code == clientDeprecationErrorKey {
+			return locale.NewError("err_buildplanner_deprecated", "Encountered deprecation error: {{.V0}}", graphqlErr.Message)
```

Comment:
Is this already not being reported to rollbar because it is a buildplan error or do we need to flag this one too?

Reply:
Yeah, we need to mark this as an input error as well. Thanks for catching that.